### ISSUE TITLE
Tasks will fail when settings.USE_TZ = False

### ISSUE THREAD

**Issue by ryanbagwell** [hzdg/django-periodically#8]
By default, the Django setting USE_TZ = False. This will throw errors when running tasks because [this reference to the current time](https://github.com/hzdg/django-periodically/blob/master/periodically/backends.py#L68) is not timezone aware.

The tests pass because they're bypassing the task running command.

I'm not sure what the best way to handle this is. My thought is that we should require USE_TZ to be True for consistency, and if it's not, throw an assertion error with a clear explanatory message.

I created an additional test in order to test the runner task. It's in a branch tz-issues.

**Comment by matthewwithanm** [comment]
According to [this SO question](http://stackoverflow.com/a/10784482/155370), `timezone.now()` is supposed to respect `USE_TZ` and return a timezone-aware datetime. Is the problem that it's always UTC?

**Comment by ryanbagwell** [comment]
From the [Django docs for timezone.now()](https://docs.djangoproject.com/en/dev/ref/utils/#django.utils.timezone.now):

`If USE_TZ is False, this will be a naive datetime (i.e. a datetime without an associated timezone) that represents the current time in the system’s local timezone.`

`If USE_TZ is True, this will be an aware datetime representing the current time in UTC. Note that now() will always return times in UTC regardless of the value of TIME_ZONE; you can use localtime() to convert to a time in the current time zone.`

By default, USE_TZ = False, which means by default, timezone.now() will not return an aware datetime.

**Comment by matthewwithanm** [comment]
Oh, so the issue is that a time generated elsewhere _is_ timezone aware, mucking up the comparison?

**Comment by ryanbagwell** [comment]
Exactly. [Here's the first culprit.](https://github.com/hzdg/django-periodically/blob/master/periodically/backends.py#L68)

**Comment by matthewwithanm** [comment]
Where's the date that one's being compared to? Also, can you paste a stacktrace?

**Comment by ryanbagwell** [comment]
The [schedule class is where the comparison object is made timezone aware](https://github.com/hzdg/django-periodically/blob/master/periodically/schedules.py#L100).

The comparison is happening in the [time_before method](https://github.com/hzdg/django-periodically/blob/master/periodically/schedules.py#L113) of the schedule class.

The `time_before()` method gets called in [`utils.get_scheduled_time()`](https://github.com/hzdg/django-periodically/blob/master/periodically/utils.py#L20) which passes the unaware datetime object.

`utils.get_scheduled_time()` is called by [`backends.BaseBackend._run_tasks()`](https://github.com/hzdg/django-periodically/blob/master/periodically/backends.py#L94), which is where the unaware datetime is created.

**Comment by matthewwithanm** [comment]
Ah, so maybe we should only make [this guy](https://github.com/hzdg/django-periodically/blob/f2e416e9332db59126bc83f815742b9363241c23/periodically/schedules.py#L100) timezone aware if `settings.USE_TZ`?
